Automattic/wp-calypso
# Add reader-join-conversation modal block

This PR will add a new dialog/modal when a logged out user clicks on a reader action (that requires the user to be logged in).

The modal will inform the user that they need to either create a new WordPress.com account or log in to an existing WordPress.com account to interact with the reader.

This PR adds tracks events;
`calypso_reader_dialog_login_clicked` - when user clicks the login link in the modal
`calypso_reader_dialog_create_account_clicked` - when the user clicks the create account button in the modal

Project thread - p2-pe7F0s-1gz

- [x] Create LoginWindow Component
- [x] Update Like button to use LoginWindow Component when clicked and user is logged out
- [x] Add feature flag to LoginWindow so we can merge without side affects
- [x] Add stats/tracks for login/create account actions from LoginWindow

### Testing (so far)

* Sandbox the domains `public-api.wordpress.com` and `wpcalypso.wordpress.com`
* Requires patch code-D124724 - this handles the postMessage to inform popup window of login event - without this, the page will not refresh after login is complete
* Open incognito window in browser and make sure you are logged out of WordPress.com
* Go to a tag page, i.e. http://calypso.localhost:3000/tag/writing
* Click the like icon  in the post stream and you should get the modal below
<img width="354" alt="Screenshot 2023-10-11 at 22 00 39" src="https://github.com/Automattic/wp-calypso/assets/5560595/14194ca4-69ec-435f-94d5-bb4e490fe24c">

* Click either the create and account button or login link
* You should see the login popup - login to your WordPress.com account
* The popup should automatically close and the reader page should refresh with you logged in

**Things to look out for**
* There has been a change to how we handle logged out likes - the functionality is now in a `onLoggedOut` function that is passed as a prop to the like container. There is a feature flag check in this function that checks to see if the loginWindow feature is enabled; if it is the new modal should show when logged out like event occurs, otherwise the user is redirected to the create account (as it happens without PR).
